internal/policies/mount/mount_test.go

```diff
@@ -0,0 +1,188 @@
+
+		// Multiple entries cases.
+		"successfully generates mounts file for multiple entries with one value":       {entries: "multiple entries with one value"},
+		"successfully generates mounts file for multiple entries with multiple values": {entries: "multiple entries with multiple values"},
+		"successfully generates mounts file for multiple entries with the same value":  {entries: "multiple entries with the same value"},
+		"successfully generates mounts file for multiple entries with repeated values": {entries: "multiple entries with repeated values"},
+
+		// Special cases.
+		"successfully generates mounts file for errored entries":   {entries: "errored entries"},
+		"successfully generates mounts file with anonymous values": {entries: "entry with anonymous tags"},
+
+		// Badly formatted entries.
+		"successfully generates mounts file trimming whitespaces":           {entries: "entry with linebreaks and spaces"},
+		"successfully generates mounts file trimming sequential linebreaks": {entries: "entry with multiple linebreaks"},
+		"generates an empty file if the entry is empty":                     {entries: "one entry with no value"},
+		"generates no file if there are no entries":                         {entries: "no entries"},
+
+		// Policy refresh.
+		"mount file is removed on refreshing policy with no entries": {secondCall: "no entries"},
+		"mount file is updated on refreshing policy with one entry":  {secondCall: "one entry with multiple values"},
+
+		// Error cases.
+		"error when user is not found":              {objectName: "dont exist", wantErr: true},
+		"error when user has invalid uid":           {userReturnedUID: "invalid", wantErr: true},
+		"error when user has invalid gid":           {userReturnedGID: "invalid", wantErr: true},
+		"error when usrDir has invalid permissions": {readOnlyUserDir: true, wantErr: true},
+
+		// To be removed when computer policies get implemented.
+		"error when trying to apply computer policies": {computer: true, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			testRunDir := t.TempDir()
+
+			opts := []mount.Option{mount.WithRunDir(testRunDir)}
+
+			usr, _ := user.Current()
+			if tc.objectName == "" {
+				if tc.userReturnedUID == "" {
+					tc.userReturnedUID = usr.Uid
+				}
+				if tc.userReturnedGID == "" {
+					tc.userReturnedGID = usr.Gid
+				}
+
+				tc.objectName = "ubuntu"
+
+				opts = append(opts, mount.WithUserLookup(func(string) (*user.User, error) {
+					return &user.User{Uid: tc.userReturnedUID, Gid: tc.userReturnedGID}, nil
+				}))
+			}
+
+			entries := mount.EntriesForTests["one entry with one value"]
+			if tc.entries != "" {
+				entries = mount.EntriesForTests[tc.entries]
+			}
+
+			if tc.readOnlyUserDir {
+				opts = append(opts, mount.WithPerm(0100))
+
+				// Ensures that the test dir will be cleaned after the test.
+				defer func(p string) {
+					//nolint:errcheck,gosec
```

Comment:
Always better to add a rationale on why we are ignoring the error :)
I would still do `_ = os.Chmod()` though